### DISCUSSION TITLE
Billing - Invoices, Xero, Rates, Stand Downs, Billable Items

### DISCUSSION THREAD

**Discussion by dylanhru** [stayonhire/feedback#16] [Upcoming Features]
Billing features are broken down into the following sub-features:

1. Invoice cycles
2. Asset rates
3. Billable items (Stand downs, freight, fuel, damages)
4. Invoice generation and sync to Xero

**Asset Rates**
An update has been released for entering rates under two types:

_Flat Rate:_
![Screenshot 2023-07-20 at 1 05 19 pm](https://github.com/stayonhire/feedback/assets/34927957/20811616-d9bc-472c-b446-023c6c62704d)

_Minimum duration:_
![Screenshot 2023-07-20 at 1 05 12 pm](https://github.com/stayonhire/feedback/assets/34927957/09a32f5a-d2cc-43f3-9040-52514f6d6a35)

**Comment by memphisplanthire** [comment]
I have been cc'ing myself in when I send invoices, but I haven't received them - just checked and the client hasn't received either. So emails are recorded as being sent but not being received (not in junk etc either)

**Reply by dylanhru** [reply]
Hi Memphis Team,

Sorry, I tried giving you a call today but must have missed you. In regards to the send functionality, I just tested it and it does currently work -though I believe your issue might be one of having a bad user experience with the component.

I suspect it could be to do with needing to press 'Enter' after typing a recipient email address in the send bottom sheet:
<img width="620" alt="Screenshot 2023-08-21 at 8 47 35 pm" src="https://github.com/stayonhire/feedback/assets/34927957/9ca430d5-ab37-44dc-a4ae-5b52b1402880">

Example test:
![IMG_E00284133358-1](https://github.com/stayonhire/feedback/assets/34927957/a57ceea7-edac-46e4-9af2-bf08a8f5fe8b)

![IMG_6157FEFEF5AB-1](https://github.com/stayonhire/feedback/assets/34927957/179141ac-e3e3-4102-97e2-c5c501aafc50)

I will give you a ring tomorrow morning to get to the bottom of it and make a plan :)

**Comment by memphisplanthire** [comment]
Hey Dylan,

Haven't heard from you yet, I've gone back in and had a look but I'm not sure if this is the issue - I haven't been "typing" the email addresses in as they are generated from the account, and even the "send to me" email isn't working either? Would it be helpful if I tried to screen record?

Thanks

[cid:a27605c3-5557-4817-9b15-a5e209eda5b1]<https://memphis-plant-hire.square.site/>
________________________________

**Reply by memphisplanthire** [reply]
Hey team, something else I've noticed. I applied a stand down day to the wrong invoice. I went in and removed it, but it's still showing as "resolved" when I need to apply it to another invoice. I also seem to have lost where to apply stand downs to invoices?
![Stayonhire](https://github.com/stayonhire/feedback/assets/142187468/b81ec51c-87a9-4d97-b2ec-03703225e701)

**Reply by dylanhru** [reply]
Hi Crystelle,

**Invoice Sending**
I may need more information and/or a video of the problem sent to me via WhatsApp. I've tested multiple scenarios over and over again and they do indeed send, both to auto-populated email addresses (coming from Xero) and manually typed ones.

**Removing Stand Downs**
I just took a look at your account and it does appear you found the "Mark as Un-Resolved" button, so I suspect this is not necessarily a problem anymore.

_Select the billable item row and mark it as un-resolved manually from the action menu:_
![Screenshot 2023-09-01 at 3 14 13 pm](https://github.com/stayonhire/feedback/assets/34927957/7de7673f-0fa8-49da-8713-462f0062648f)

**Comment by memphisplanthire** [comment]
Hi team,

Another thing, I noticed the ABN was wrong on the system, for some reason our old ABN was on there. I've corrected it in the system, but the invoice that I'd already issued still has the old ABN. Is there a way to update this? Thanks

**Reply by dylanhru** [reply]
Sorry about that! I must have got it from an ABN lookup when setting up the demo account.

I presume this means the invoice is in an approved state. If so, re-drafting the invoice would be the solution. 

All newly drafted invoices will contain your updated account details.

Other invoice template tweaks can also be made from **Settings > Documents > Invoices** :
![Screenshot 2023-09-01 at 3 25 38 pm](https://github.com/stayonhire/feedback/assets/34927957/e8fc3a23-c925-4635-8619-e8cd8a37c077)

**Reply by dylanhru** [reply]
Hi @memphisplanthire,

Do you want to schedule time with me [(View Calendar)](https://calendly.com/dylan-hrurukvit) sometime and we can go through recent updates to Billing? 

There have been so many over Oct/Nov 2023 already and I would love to confirm everything is coming along as you'd ideally want :)

**Comment by WadeRolfe** [comment]
Hi team,

Is there any updates on adding a weekly billing rate into the items?

**Reply by dylanhru** [reply]
Hi @WadeRolfe, I've created a seperate upcoming feature post to describe the details of weekly rates, as it's a little more involved than a simple reply in this chain. 

Can you and Kaylen have a look at the following scope and let me know if there is anything else you'd need for ABP?

#43[ Base Category Rates (Daily, Weekly, Monthly etc.)](https://github.com/orgs/stayonhire/discussions/43)

Thanks!

**Comment by dylanhru** [comment]
Billing permissions have been added to Settings > Users.

This will prevent un-authorised users from seeing anything other than the asset rate.

Details have bee added to #42 

> A new billing permissions toggle has been added to Settings > Users.
> 
> This will hide all billing, invoicing and payment information from users who are not authorised.
> 
> ![Screenshot 2023-11-09 at 10 46 39 am](https://user-images.githubusercontent.com/34927957/281600613-d33ebcca-7acd-493c-9a51-f9343441fbd2.png)
> 
> The 'Rate' cell in Job Cards is only remaining field that is not hidden from 'non-authorised users.
> 
> ![Screenshot 2023-11-09 at 10 58 25 am](https://user-images.githubusercontent.com/34927957/281601957-92bc48ce-0028-407c-8abd-be68d983e51a.png)
